rfcs/0001-restructure.md

```diff
@@ -0,0 +1,271 @@
+  mysqli, openssl, phpiredis, zlib etc; and a suitable `php.ini`. Downstream
+  buildpacks are expected to obtain info (like extension dir) through the
+  [`php-config`](https://www.php.net/manual/en/install.pecl.php-config.php)
+  executable whenever possible.
+
+* **composer**:
+  Installs [`composer`](https://getcomposer.org), a dependency manager for PHP
+  and makes it available on the `$PATH`
+  * provides: `composer`
+  * requires: none
+
+* **composer-install**:
+  Resolves project dependencies, and installs them using `composer`.
+  * provides: none
+  * requires: `php`, `composer` at `build`
+
+* **php-fpm**:
+  Configures `php-fpm.conf` (config file in `php.ini` syntax), and sets a start
+  command (type `php-fpm`) to start FPM[<sup>1</sup>](#note-1).
+  * provides: `php-fpm`
+  * requires: `php` during launch
+
+  Separation of FPM into a separate buildpack lets users run FPM in one
+  container and web server in another container.
+  The generated `php-fpm.conf` must accept requests on a tcp/ip socket by
+  default, must "include" custom fpm configs specified by the user (e.g. in
+  buildpack.yml or specific location in the app).
+
+* **php-builtin-server**:
+  Set up PHP's [built-in web
+  server](https://www.php.net/manual/en/features.commandline.webserver.php) to
+  serve PHP applications.
+  * provides: none
+  * requires: `php` at `launch`
+
+  This buildpack sets a start command (type `web`) to start the built-in web
+  server. This is the default web server.
+
+* **php-httpd**:
+  Sets up HTTPD as the web server to serve PHP applications.
+  * provides: none
+  * requires: `php`, `php-fpm`, `httpd`, at `launch`
+
+  This buildpack generates `httpd.conf` and sets up a start command (type
+  `web`) to run PHP FPM and HTTPD Server. Apps need to declare the intention to
+  use httpd in `buildpack.toml`:
+
+* **php-nginx**:
+  Sets up Nginx as the web server to serve PHP applications.
+  * provides: none
+  * requires: `php`, `php-fpm`, `nginx` at `launch`
+
+  This buildpack generates `nginx.conf` and sets up a start command (type
+  `web`) to run PHP FPM and Nginx Server. Apps need to declare the intention to
+  use nginx in `buildpack.toml`:
+
+* **php-memcached-session-handler**:
+  Configures the given memcached service instance as a PHP session
+  handler[<sup>3</sup>](#note-3). Memcached settings are to be provided through
+  a suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+* **php-redis-session-handler**:
+  Configures the given redis service instance as a PHP session
+  handler[<sup>3</sup>](#note-3). Redis settings are to be provided through a
+  suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+This would result in the following order groupings in the PHP language family meta-buildpack:
```

Comment:
As discussed in the working group meeting, we'll need to cover the use case for stand-alone PHP cli scripts. Using Procfile would make the most sense and would keep things consistent with other buildpacks.

Reply:
I can think of 2 scenarios:

A) Force user to provide a Procfile if they wanna run a php cli script. Default behavior of the meta-buildpack would be to run the built-in server.
B) Builtin server would only be run if a certain directory structure (like `<app>/htdocs` - we do this now) is maintained. Default behavior of the meta-buildpack would be to run the php script.

What do you think meets users default expectation better?

Reply:
Updated RFC with option (A)

Reply:
I think the behavior when it's not a web app and no Procfile is present should be to fail, with some sort of message like `WARNING: Buildpack doesn't know how to run this app. If this is a web app, make sure you've set a web dir and that it exists. If this is a PHP cli application, you'll need to provide a Procfile to indicate the command to execute.`